Lib/test/test_genericalias.py

```diff
@@ -176,6 +176,19 @@ class L(list): ...
         with self.assertRaises(TypeError):
             issubclass(L, list[str])
 
+    def test_type_generic(self):
+        t = type[int]
+        Test = t('Test', (), {})
+        self.assertTrue(isinstance(Test, type))
+        test = Test()
+        self.assertEqual(t(test), Test)
+
+    def test_type_subclass_generic(self):
+        class MyType(type):
+            pass
+        with self.assertRaises(TypeError):
+            MyType[int]
```

Comment:
Interestingly, this is different from `test_generic_subclass` (at line 96 above).

I'm okay with that, but I wonder if you're just testing an implementation accident here? Else what's the reason we want `MyType[int]` to fail, but we want `MyList[int]` to work?

Reply:
I believe `type` is special in this case. Generic containers like `list[str]` are type constructors that build `list`s. It would make sense `MyList[str]` would make `MyList`s. However, with `type`, there is only `typing.Type`, there is no other way to construct the `type` of an instance. It doesn't (AFAIK) make any sense to allow `MyType[SomeClass]`.

I can remove this test if that seems unreasonable.